microprofile/WFLY-13640_MicroProfile_Reactive_Messaging.adoc

```diff
@@ -0,0 +1,203 @@
+
+* In memory - this is not actually a connector, it is part of the core SmallRye Reactive Messaging provider
+* Kafka - used to interact with Kafka/AMQ Streams
+
+[[spec-api]]
+==== Spec API
+With the way SmallRye Reactive Messaging is implemented, we cannot rely on the Reactive Messaging 1.0 API jar at runtime. This is because SmallRye Reactive Messaging has its own API ar which is used internally and contains a copy of the spec package. This copy of the https://github.com/smallrye/smallrye-reactive-messaging/tree/master/api/src/main/java/org/eclipse/microprofile/reactive/messaging[spec package] has some changes compared with the
+https://github.com/eclipse/microprofile-reactive-messaging/tree/1.0/api/src/main/java/org/eclipse/microprofile/reactive/messaging[MicroProfile Reactive Messaging 1.0 spec API].
+These changes are:
+<<<<<<< HEAD
+
+* The `Message` interface has new methods, and additionally a bunch of new factory methods (used both internally, and for users to wrap the payload in a `Message` instance from their `@Outgoing` annotated methods).
+* New classes:
+** `@Channel` annotation
+** `Emitter` interface - Only really makes sense if annotated with `@Channel`
+** `@OnOverflow` annotation
+** `@Metadata` class - Used by some of the new `Message` methods
+
+=======
+
+* The `Message` interface has new methods, and additionally a bunch of new factory methods (used both internally, and for users to wrap the payload in a `Message` instance from their `@Outgoing` annotated methods).
+* New classes:
+** `@Channel` annotation
+** `Emitter` interface - Only really makes sense if annotated with `@Channel`
+** `@OnOverflow` annotation
+** `@Metadata` class - Used by some of the new `Message` methods
+
+>>>>>>> b7a909b... Reduce scope for RM 1.0 and Kafka
+
+These changes are backward compatible, so the best we can do right now is to have users compile against the Reactive Messaging 1.0 API to 'hide' the new functionality from them. These will be included in the BOMs.
+
+Additionally, a deployment unit processor in the Reactive Messaging subsystem will by default ensure that there are no methods annotated with `@Channel` and `@Emitter` and throw an error if someone tries to use them from their deployment (for example if they compiled against the SmallRye Reactive Messaging API jar insteead of against the MicroProfile). The same applies to
+the usage of annotations from the `io.smallrye.reactive.messaging.annotations` package which are also considered experimental. Note that in WildFly, use of these will be allowed if the `jboss.as.reactive.messaging.experimental` system property is set to
+`true`, as this allows us to test the full functionality of each SmallRye Reactive Messaging upgrade. For product, this allowing
+of experimental features will be totally disabled.
+
+
+
+=== Organisation
+There is an extension/subsystem implementing each of the three specifications mentioned:
```

Comment:
There are mentioned following specifications here:

* https://github.com/eclipse/microprofile-reactive-messaging/releases - Reactive Messaging 1.0
* https://github.com/eclipse/microprofile-reactive-streams-operators/releases - RSO 1.0.1

what is the third one?

Reply:
That is Context Propagation, which is no longer part of this. It lives in the [feature pack](https://github.com/wildfly-extras/wildfly-mp-reactive-feature-pack) - the `development` branch contains the bits which have not yet 'incubated' to WildFly.

I will change `three` to `two`